### PR TITLE
=per #18123 Deprecate PersistentView

### PR DESCRIPTION
**Pull request by patriknw**
- marked PersistentFSM as experimental
- also changed the order of some sections in migration guide
- link to 2.2->2.3 migration guide
- update experimental index page

### CODE REVIEW

File: akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst [hunk 1]
```diff
@@ -30,21 +30,11 @@ in practice. No changes were needed to the Akka source code for this update. Use
 depend on 3.8.0.Final that break with 3.10.3.Final should be able to manually downgrade the dependency
 to 3.8.0.Final and Akka will still work with that version.
 
-Added parameter validation to RootActorPath
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst]: Only restructuring in this file, except for the new section "PersistentView is deprecated"

File: akka-docs/rst/experimental/index-java.rst [hunk 2]
```diff
@@ -19,10 +19,10 @@ prior deprecation.
 .. toctree::
    :maxdepth: 1
 
-   ../java/persistence
    ../dev/multi-node-testing
    ../java/lambda-actors
    ../java/lambda-fsm
+   ../java/persistence-query
```
Comment (ktoso) [inline akka-docs/rst/experimental/index-java.rst]: Whoops, so I've put it in the wrong section: https://github.com/akka/akka/blame/master/akka-docs/rst/java/index-actors.rst#L15
Would you remove it from there in this PR as well?

Comment (ktoso) [inline akka-docs/rst/experimental/index-java.rst]: What about https://github.com/akka/akka/blob/master/akka-docs/rst/experimental/index.rst updated last 2 years ago hmmm

File: akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst [hunk 3]
```diff
@@ -386,6 +386,26 @@ have explicitly relied on this behaviour, however if you find yourself with an a
...
+  being propagated all the way
+* `ActorSubscriber`_ in case you need more fine grained control
+  
+The consuming actor may be a plain ``Actor`` or a ``PersistentActor`` if it needs to store its
+own state (e.g. fromSequenceNr offset).
+
+.. _Sink.actorRef: http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-integrations.html#Sink_actorRef
+.. _mapAsync: http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/stages-overview.html#Asynchronous_processing_stages
+.. _ActorSubscriber: http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-integrations.html#ActorSubscriber
+
```
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.3.x-2.4.x.rst]: Very good summary of options, (and :+1: on including the entire text in the docs above)

**Comment by ktoso** [comment]
LGTM

File: akka-docs/rst/experimental/index-java.rst (hunk 2 above)
Comment (patriknw) [inline akka-docs/rst/experimental/index-java.rst]: it's fine to have it in both places, same for Akka Typed

**Comment by ktoso** [comment]
Refs https://github.com/akka/akka/issues/18123 and needs a rebase sadly

**Pull request merged**